WGS_pipeline/PureCN_pipeline/PureCN.wdl

```diff
@@ -0,0 +1,112 @@
+task run_PureCN {
+
+  # File-related inputs
+  String sampleID
+  File segFile
+  File vcf
+  File intervals
+  File call_wgd_and_cin_script
+
+  # Method configuration inputs
+  String genome
+  Int? maxCopyNumber
+  Float? minPurity
+  Float? maxPurity
+  String? funSegmentation
+  Int? maxSegments
+  String? otherArguments
+
+  # Hardware-related inputs
+  Int? hardware_disk_size_GB = 50
+  Int? hardware_memory_GB = 4
+  Int? hardware_preemptible_tries = 2
+
+  command {
+    Rscript /opt/PureCN/PureCN.R \
+      --out /cromwell_root/"${sampleID}" \
+      --sampleid "${sampleID}" \
+      --seg-file "${segFile}" \
+      --vcf "${vcf}" \
+      --intervals "${intervals}" \
+      --genome "${genome}" \
+      ${"--max-purity " + maxPurity} \
+      ${"--min-purity " + minPurity} \
+      ${"--max-copy-number " + maxCopyNumber} \
+      ${"--fun-segmentation " + funSegmentation} \
+      ${"--max-segments " + maxSegments} \
+      ${otherArguments}
+    Rscript -e "write.table(read.csv('${sampleID}.csv'),'table.txt',sep='\n',row.names=F,col.names=F,quote=F)"
+    Rscript ${call_wgd_and_cin_script} "${sampleID}_loh.csv" "${sampleID}.csv"
+  }
+
+  output {
+    File solutions_pdf = "${sampleID}.pdf"
+    File chromosomes_pdf = "${sampleID}_chromosomes.pdf"
+    File rds = "${sampleID}.rds"
+    File dnacopy = "${sampleID}_dnacopy.seg"
+    File variants = "${sampleID}_variants.csv"
+    File loh = "${sampleID}_loh.csv"
+    File genes = "${sampleID}_genes.csv"
+    File segmentation = "${sampleID}_segmentation.pdf"
+    File log = "${sampleID}.log"
+    File selected_solution = "${sampleID}.csv"
+    File local_optima_pdf = "${sampleID}_local_optima.pdf"
+    Array[String] table = read_lines("table.txt")
+    String purity = table[1]
+    String ploidy = table[2]
+    String contamination = table[4]
+    String flagged = table[5]
+    String curated = table[7]
+    String comment = table[8]
+    Array[String] wgd_table = read_lines("out.txt")
+    String wgd = wgd_table[0]
+    String loh_fraction = wgd_table[1]
+    String cin = wgd_table[2]
+    String cin_allele_specific = wgd_table[3]
+    String cin_ploidy_robust = wgd_table[4]
+    String cin_allele_specific_ploidy_robust = wgd_table[5]
+  }
+
+  runtime {
+    docker: "markusriester/purecn:latest"
```

Comment:
never use latest

Reply:
let the version be a parameter

Reply:
I fixed this on dev but forgot to update it here. done now!